pocs/tests/test_horizon_limits.py

```diff
@@ -0,0 +1,263 @@
+def validation_2(obstruction_points):
+    for i in obstruction_points:
+        assert type(i) == tuple
+        assert len(i) == 2
+
+
+#       If any point is the incorrect data type
+def validation_3(obstruction_points):
+    assert type(obstruction_points) == list
+    for i in obstruction_points:
+        assert type(i[0]) == float or type(i[0]) == int
+        assert type(i[1]) == float or type(i[1]) == int
+
+
+#       If any point isnt in the azimuth range (0<=azimuth<359,59,59)
+def validation_4(obstruction_points):
+    for i in obstruction_points:
+        az = i[0]
+        assert az >= 0 and az < 360  # degrees are astropy units degrees
+
+
+#       If any point isnt in the elevation range (0<=elevation<90)
+def validation_5(obstruction_points):
+    for i in obstruction_points:
+        el = i[1]
+        assert el >= 0 and el <= 90
+
+
+#       If any inputted azimuth isnt in the correct sequence low and increasing order
+def validation_6(obstruction_points):
+    az_list = []
+    for i in obstruction_points:
+        az_list.append(i[0])
+    assert sorted(az_list) == az_list
+
+
+#       If multiple data points have the same azimuth thrice or more
+#       This works assuming that the array is sorted as per validation_6
+def validation_7(obstruction_points):
+    azp1 = -1
+    azp2 = -1
+    for i in obstruction_points:
+        az = i[0]
+        assert az != azp1 or azp1 != azp2
+        azp2 = azp1
+        azp1 = az
+
+
+def obstruction_points_valid(obstruction_points):
+    v = True
+    try:
+        validation_1(obstruction_points)
+    except AssertionError:
+        print("obstruction_points should have at least 2 sets of points")
+        v = False
+    try:
+        validation_2(obstruction_points)
+    except AssertionError:
+        print("Each point should have 2 of components (az and el")
+        v = False
+    try:
+        validation_3(obstruction_points)
+    except AssertionError:
+        print("obstruction_points should be a list of tuples where each element is a float or an int")
+        v = False
+    try:
+        validation_4(obstruction_points)
+    except AssertionError:
+        print("obstruction_points should be in the azimuth range of 0 <= azimuth < 60")
+        v = False
+    try:
+        validation_5(obstruction_points)
+    except AssertionError:
+        print("obstruction_points should be in the elevation range of 0 <= elevation < 90")
+        v = False
+    try:
+        validation_6(obstruction_points)
+    except AssertionError:
+        print("obstruction_points azimuth's should be in an increasing order")
+        v = False
+    try:
+        validation_7(obstruction_points)
+    except AssertionError:
+        print("obstruction_points should not consecutively have the same azimuth thrice or more")
+        v = False
+    assert v
+    return v
+
+
+# Unit tests for each of the evaluations
+def test_validations():
```

Comment:
I'm a bit confused as to what is going on here. The `validation_x` subs should be given names that make sense. If `validation_1` is testing that there is only one point it should be called `test_single_point` and a `test_no_points_fail` or something like that. Maybe we can sit down and go over a little bit of this in person.